### PR TITLE
Add CICD for build and test

### LINKED ISSUE #2: Add basic CI/CD tests
* Verify that things compile
* Add a CI/CD test to test that the ebpf program passes verification

We should add at least minimal CI/CD tests to prevent regressions

### LINKED ISSUE #3: Create a test .o file for Cilium demo
`ebpf-for-windows-demo.sln` should also generate "test" .o files for both SNAT and DSR scenario, which can be used for testing / verification purpose.

### PR DESCRIPTION
**Pull request by saxena-anurag**
This PR does the following:

1. Adds CICD to perform daily and _on-pull-request_ build and test runs.
2. The solution now builds .o files for SNAT and DSR
3. Added tests to run verifier on the .o files
4. Added project to build ebpf-for-windows solution

Fixes #2 
Fixes #3

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -1,39 +1,107 @@
...
+        7z x LLVM10.exe -y -o"C:/Program Files/LLVM"
+        echo "C:\Program Files\LLVM\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
 
     - name: Restore NuGet packages
       working-directory: ${{env.GITHUB_WORKSPACE}}
-      run: nuget restore ${{env.SOLUTION_FILE_PATH}}
+      run: |
+        cd external\ebpf-for-windows
+        nuget restore ${{env.EBPF_FOR_WINDOWS_SOLUTION_FILE}}
+        cd ..\..
```
Comment (dthaler) [inline .github/workflows/cicd.yml:65]: I am not sure you need the "cd ..\.." at the end of a step anywhere since each step has its own "working-directory" directive.

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -31,25 +31,16 @@ The following steps need to be executed _once_ before the first build on a new c
 3. ```cmake -S external\ebpf-verifier -B external\ebpf-verifier\build```
 4. ```nuget restore ebpf-for-windows.sln```
 5. ```del external\ebpf-verifier\build\obj\project.assets.json```
+6. Change directroy back to the solution root, e.g. ```cd ..\..\```.
```
Comment (dthaler) [inline docs/GettingStarted.md:34]:
```suggestion
6. Change directory back to the solution root, e.g. ```cd ..\..\```.

```

File: ebpf-for-windows/ebpf-for-windows.vcxproj [hunk 3]
```diff
@@ -0,0 +1,187 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (dthaler) [inline ebpf-for-windows/ebpf-for-windows.vcxproj:1]: Why is this file needed in this repo?  It should already be in the ebpf-for-windows project.  Or is this a placeholder until we have a devkit you can install instead of cloning/building the full ebpf-for-windows repo?

File: .gitmodules [hunk 4]
```diff
@@ -5,3 +5,6 @@
 [submodule "external/ebpf-for-windows"]
 	path = external/ebpf-for-windows
 	url = https://github.com/microsoft/ebpf-for-windows.git
+[submodule "external/Catch2"]
+	path = external/Catch2
+	url = https://github.com/catchorg/Catch2.git
```
Comment (dthaler) [inline .gitmodules:10]: ebpf-for-windows already has catch2 as a submodule.  Can't you just use external\ebpf-for-windows\external\catch2 without adding it again here?

File: tests/include/catch_wrapper.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation
```
Comment (dthaler) [inline tests/include/catch_wrapper.hpp:1]: Can you just use external\ebpf-for-windows\tests\libs\util\catch_wrapper.hpp instead of adding this file here?

File: ebpf-for-windows/ebpf-for-windows.vcxproj (hunk 3 above)
Comment (saxena-anurag) [reply ebpf-for-windows/ebpf-for-windows.vcxproj:1]: `ebpf-for-windows` submodule has a solution file. This project is added to perform a custom build step (executes msbuild for `ebpf-for-windows` solution).

I could not find any good way to include a solution file to another solution file. I could have included the individual projects present in `ebpf-for-windows` sln into the top-level solution in this repo, but that will not be scalable.

File: tests/include/catch_wrapper.hpp (hunk 5 above)
Comment (saxena-anurag) [reply tests/include/catch_wrapper.hpp:1]: Done.

File: .gitmodules (hunk 4 above)
Comment (saxena-anurag) [reply .gitmodules:10]: Actually the commit of ebpf-for-windows repo that this repo is pointing to does not have catch2 as a submodule. I can fix this when I update the ebpf-for-windows submodule to latest.

Comment (saxena-anurag) [reply .gitmodules:10]: Filed Issue #8 to track this.

**Review by dthaler — APPROVED** [review]

File: ebpf-for-windows-demo.sln [hunk 6]
```diff
@@ -9,37 +9,129 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "daemon", "cilium\load_balan
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "bpf_xdp", "cilium\load_balancer\bpf\bpf_xdp\bpf_xdp.vcxproj", "{742B767C-02C3-4D5A-81A9-232D57262388}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{9171E26A-58B7-4DAB-B00B-C9F0F667D6D5}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "cilium_test", "tests\cilium_tests\cilium_test.vcxproj", "{12161211-5463-4AA9-9B6E-552552B475DC}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Catch2WithMain", "external\Catch2\build\src\Catch2WithMain.vcxproj", "{8BD3552A-2CFB-4A59-AB15-2031B97ADA1E}"
```
Comment (shankarseal) [inline ebpf-for-windows-demo.sln:16]: if you are using catch as  a submodule and linking as a lib, then you dont need nuget package.config for catch.

**Review by shankarseal — APPROVED** [review]

File: ebpf-for-windows-demo.sln (hunk 6 above)
Comment (saxena-anurag) [reply ebpf-for-windows-demo.sln:16]: Updated issue #8 to fix this issue. Currently it is not possible to fix this as demo repo is including catch_wrapper.hpp from ebpf-for-windows submodule, and the submodule is pointing to an older commit.

**Review by shankarseal — APPROVED** [review]
